src/SignalR/server/Core/src/HubConnectionHandler.cs

```diff
@@ -226,7 +226,7 @@ private async Task SendCloseAsync(HubConnectionContext connection, Exception? ex
 
             try
             {
-                await connection.WriteAsync(closeMessage);
+                await connection.WriteCloseAsync(closeMessage);
```

Comment:
I prefer adding an `ignoreAbort = false` parameter to WriteAsync. Where it checks `if (_connectionAborted)` today, we'd change that to `if (_connectionAborted && !ignoreAbort)`. That way if we make a change to WriteAsync in the future, we get it for free without having to remember to also update WriteCloseAsync.

Reply:
and add a branch to every single write!? 😮

Reply:
But the branch is already there 😆